lib/kb/models/concerns/searchable.rb

```diff
@@ -0,0 +1,19 @@
+module KB
+  module Searchable
+    extend ActiveSupport::Concern
+
+    included do
+      include Queryable
+    end
+
+    module ClassMethods
+      def search(filters)
+        response = kb_client.request('search', filters: filters).symbolize_keys
+        elements = response[:elements].map { |contract| from_api(contract) }
+        response.merge(elements: elements)
```

Comment:
What do you think of wrapping the result in some SearchResult object? It's not really convenient to work with raw hashes and access elements as `search[:elements]`.

Reply:
I thought of that wrapping object before, but I wasn't sure about what would be a good design for it, as it is essentially subset of items (not necessarily obtained from a search) with pagination metadata. I also thought of making it an `Enumerable` with custom metadata.
So, in order to not overthink stuff nor implementing the wrong thing, I stuck to a simple hash structure.
If you have a clear idea about how it should look and feel sure about it, I'll be happy to give it a go.

Reply:
> but I wasn't sure about what would be a good design for it, as it is essentially subset of items (not necessarily obtained from a search) with pagination metadata.

Not sure I understand the confusion. I don't see any design decision to make here. I only wanted to avoid working with hash directly and have a simplistic interface for the search result with `page`, `total` and `elements` methods :) 

> So, in order to not overthink stuff nor implementing the wrong thing, I stuck to a simple hash structure.
If you have a clear idea about how it should look and feel sure about it, I'll be happy to give it a go.

It seems like you gave it far more thought that I did 😅 I had something like this in mind:

```ruby

class SearchResult
  attr_reader :elements, :total, :page

  def initialize(elements:, total:, page:)
    @elements = elements
    @total = total
    @page = page
  end
end
```

having an interface allows us to change the implementation however we want.

Reply:
I'm not sure if accessing data via methods instead of keys justifies a new class at this point.
Anyway, I don't think it's worth discussing further - I'll follow your advice just in case I'm being too narrow-minded and there are other benefits I'm unable to see now.

Reply:
> I'm not sure if accessing data via methods instead of keys justifies a new class at this point.

You can go with a simple `OpenStruct` if you're not comfortable with adding a class. But I don't see any harm with having a simple class.

> there are other benefits I'm unable to see now.

The only benefit is giving us encapsulation of implementation details.